snuba/subscriptions/data.py

```diff
@@ -83,9 +101,20 @@ class SubscriptionData(ABC):
     metadata: Mapping[str, Any]
     tenant_ids: Mapping[str, Any] = field(default_factory=lambda: dict())
 
-    @abstractmethod
     def validate(self) -> None:
-        raise NotImplementedError
+        if self.time_window_sec < 60:
+            raise InvalidSubscriptionError(
+                "Time window must be greater than or equal to 1 minute"
+            )
+        elif self.time_window_sec > 60 * 60 * 24:
+            raise InvalidSubscriptionError(
+                "Time window must be less than or equal to 24 hours"
+            )
+
+        if self.resolution_sec < 60:
```

Comment:
also validate that there is not groupby in the request too

Reply:
This validation function runs for both SnQL and RPCs, I'll be updating the RPC validation to be more robust
https://github.com/getsentry/snuba/pull/6571/files#diff-8e3672e3fdccd3ce17c850cd47dfe0b41f53feabd7dcf550c30077fe3f7e31fbR154-R158
I'm planning to do that in a follow up, if that's okay. I want to validate there are no group bys, orderbys, only a single aggregation, single project id, a bunch of stuff

Reply:
Ended up adding it here! I expected to add more validation, but there wasn't much else I could think of at the moment https://github.com/getsentry/snuba/pull/6571/commits/83f1693dcb4fe17cf3c46fb8f7c4027ec222939b